libimage/pull.go

```diff
@@ -321,7 +316,14 @@ func (r *Runtime) copyFromDefault(ctx context.Context, ref types.ImageReference,
 	}
 
 	_, err = c.Copy(ctx, ref, destRef)
-	return []string{imageName}, err
+	if err != nil {
+		return nil, fmt.Errorf("unable to perform copy: %w", err)
+	}
+	image, _, err := r.LookupImage(imageName, nil)
```

Comment:
Doesn’t this also have the same problem as registry pulls? Using `ResolvedReference` might help.

(And would it pay to share the code with registry pulls? No idea.)